fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/OcflPersistentStorageUtils.java

```diff
@@ -180,9 +181,9 @@ public static String resolveExtensions(final String subPath, final boolean isRdf
      * @param fedoraIdentifier The fedora identifier
      * @return The resolved topic
      */
-    public static  String resolveTopic(final String fedoraIdentifier) {
-        if (fedoraIdentifier.endsWith(FEDORA_METADATA_SUFFIX)) {
-            return fedoraIdentifier.substring(0, fedoraIdentifier.indexOf(FEDORA_METADATA_SUFFIX));
+    public static FedoraId resolveTopic(final FedoraId fedoraIdentifier) {
+        if (fedoraIdentifier.isDescription()) {
+            return FedoraId.create(fedoraIdentifier.getBaseId());
```

Comment:
`asBaseId()`?

Reply:
I left this one because I actually fixed it in my next PR. Let me know if you want it fixed in this PR, and I'll do some git gymnastics to get it fixed here. Otherwise, the next PR will take care of it.